release.js

```diff
@@ -104,17 +89,63 @@ async function amendVersionsFile(key, version) {
   }
 }
 
+async function versionPrompt(currentVersion) {
+  const major = semver.inc(currentVersion, 'major');
+  const minor = semver.inc(currentVersion, 'minor');
+  const patch = semver.inc(currentVersion, 'patch');
+
+  const questions = [
+    {
+      type: 'list',
+      name: 'version',
+      message: 'What version do you want to release?',
+      choices: [
+        {
+          major,
+        },
+        {
+          minor,
+        },
+        {
+          patch,
+        },
+      ].map(i => {
+        const key = Object.keys(i)[0];
+        return {
+          key,
+          name: `${key} (${i[key]})`,
+          value: `${i[key]}`,
+        };
+      }),
+    },
+  ];
+  return await inquirer.prompt(questions);
+}
+
 async function release() {
   try {
     childProcess.execSync('./gradlew :Backpack:checkMavenCredentials');
-    const { version } = await inquirer.prompt(questions);
+    const packageResult = await inquirer.prompt(package_questions);
+    const packageType = packageResult.type
+    const packageVersion = (packageType == android) ? versions.android : versions.compose
+    const { version } = await versionPrompt(packageVersion);
     const commonVersion = semver.inc(versions.common, 'minor')
 
-    await amendGradleFiles('BpkAndroidVersion', version);
+    if (packageType == android) {
```

Comment:
just a bunch of if checks depending on which package to publish. not ideal, but it'll do for now

Reply:
You could always extract it to a function given it looks to be doing the same thing for each version and then you can just pass in the Name, type and version?